File: Example/iOS Example/UserIDViewController.m
//
//  UserIDViewController.m
//  PiwikTracker
//
//  Created by Mattias Levin on 16/10/14.
//  Copyright (c) 2014 Mattias Levin. All rights reserved.
//

#import "UserIDViewController.h"
#import "PiwikTracker.h"


@interface UserIDViewController ()
@end


@implementation UserIDViewController

- (void)viewDidLoad {
  [super viewDidLoad];
  
  [self toggleState];
}

- (void)toggleState {
  
  self.userIDTextField.text = [PiwikTracker sharedInstance].userID;
  
  NSString *userID = [PiwikTracker sharedInstance].userID;
  if (userID && userID.length > 0) {
    self.signinButton.enabled = NO;
    self.singoutButton.enabled = YES;
  } else {
    self.signinButton.enabled = YES;
    self.singoutButton.enabled = NO;
  }
  
}


- (IBAction)signinAction:(id)sender {
  NSLog(@"Sign in");
  
  if (self.userIDTextField.text && self.userIDTextField.text.length > 0) {
    [PiwikTracker sharedInstance].userID = self.userIDTextField.text;
    [self toggleState];
  }
  
}


- (IBAction)signOutAction:(id)sender {
  NSLog(@"Sign out");
  
  [PiwikTracker sharedInstance].userID = nil;
  [self toggleState];
  
}


@end

File: Example/ios/iOS Example/UserIDViewController.swift
import UIKit
import MatomoTracker

class UserIDViewController: UIViewController {
    @IBOutlet weak var userIDTextField: UITextField!
    @IBOutlet weak var signinButton: UIButton!
    @IBOutlet weak var signoutButton: UIButton!
    
    @IBOutlet weak var visitorIdTextField: UITextField!
    @IBAction func signinAction(_ sender: UIButton) {
        if (self.userIDTextField.text != nil) && (self.userIDTextField.text?.count)! > 0 {
            MatomoTracker.shared.userID = self.userIDTextField.text
            updateUserIdState()
        }
    }
    
    @IBAction func signOutAction(_ sender: UIButton) {
        MatomoTracker.shared.userID = nil
        updateUserIdState()
    }
    
    @IBAction func visitorIdTapped(_ sender: Any) {
        if (self.visitorIdTextField.text != nil) && (self.visitorIdTextField.text?.count)! > 0 {
            MatomoTracker.shared.forcedVisitorID = self.visitorIdTextField.text
            updateVisitorIdState()
        }
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        updateUserIdState()
        updateVisitorIdState()
    }
    
    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
        MatomoTracker.shared.track(view: ["menu","user id"])
    }
    
    private func updateUserIdState() {
        self.userIDTextField.text = MatomoTracker.shared.userID
        
        self.signinButton.isEnabled = !isVisitorIdValid()
        self.signoutButton.isEnabled = !self.signinButton.isEnabled
    }
    
    private func updateVisitorIdState() {
        self.visitorIdTextField.text = MatomoTracker.shared.forcedVisitorID
    }
    
    private func isVisitorIdValid() -> Bool {
        let currentVisitorId = MatomoTracker.shared.userID

        return (currentVisitorId != nil) && (currentVisitorId?.count)! > 0
    }
}
